riscv/PC_mux.py

```diff
@@ -0,0 +1,77 @@
+from myhdl import always_comb, block, modbv, concat, Signal
+
+from riscv.control_constants import *
+from riscv.opcode_constants import XPR_LEN
+
+
+@block
+def PC_mux(PC_src_sel, inst_DX, rs1_data, PC_IF, PC_DX,
+           handler_PC, epc, PC_PIF):
+    """
+    PC multiplexer module
+
+    :param Signal PC_src_sel : Select signal
+    :param Signal inst_DX: Instruction
+    :param Signal rs1_data: data from register source 1
+    :param Signal PC_IF: Program counter IF past
+    :param Signal PC_DX: Program counter DX
+    :param Signal handler_PC: Program counter handler
+    :param Signal epc: Exception Program Counter
+    :param Signal PC_PIF: Program counter IF current
+    """
+
+    base = Signal(modbv(0)[XPR_LEN:])
+    offset = Signal(modbv(0)[XPR_LEN:])
+
+    @always_comb
+    def PC_mux_assign():
+
+        imm_b = modbv(0)[XPR_LEN:]
```

Comment:
I would suggest using intbv unless vscale explicitly does wrap around for a particular signal.